File: app/src/test/java/fr/free/nrw/commons/UtilsTest.java
package fr.free.nrw.commons;

import org.junit.Assert;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;

public class UtilsTest {
    @Test public void stripLocalizedStringPass() {
        Assert.assertThat(Utils.stripLocalizedString("Hello"), is("Hello"));
    }

    @Test public void stripLocalizedStringJa() {
        Assert.assertThat(Utils.stripLocalizedString("\"こんにちは\"@ja"), is("こんにちは"));
    }

    @Test public void capitalizeLowercase() {
        Assert.assertThat(Utils.capitalize("hello"), is("Hello"));
    }

    @Test public void capitalizeFullCaps() {
        Assert.assertThat(Utils.capitalize("HELLO"), is("HELLO"));
    }

    @Test public void capitalizeNumbersPass() {
        Assert.assertThat(Utils.capitalize("12x"), is("12x"));
    }

    @Test public void capitalizeJaPass() {
        Assert.assertThat(Utils.capitalize("こんにちは"), is("こんにちは"));
    }
}

File: app/src/test/kotlin/fr/free/nrw/commons/UtilsTest.kt
package fr.free.nrw.commons

import org.junit.Test
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.CoreMatchers.equalTo
import java.util.*

class UtilsTest {
    @Test
    fun wikiLovesMonumentsYearBeforeSeptember() {
        val cal = Calendar.getInstance()
        cal.set(2022, Calendar.FEBRUARY, 1)
        assertThat(2021, equalTo( Utils.getWikiLovesMonumentsYear(cal)))
    }

    @Test
    fun wikiLovesMonumentsYearInSeptember() {
        val cal = Calendar.getInstance()
        cal.set(2022, Calendar.SEPTEMBER, 1)
        assertThat(2022, equalTo( Utils.getWikiLovesMonumentsYear(cal)))
    }

    @Test
    fun wikiLovesMonumentsYearAfterSeptember() {
        val cal = Calendar.getInstance()
        cal.set(2022, Calendar.DECEMBER, 1)
        assertThat(2022, equalTo( Utils.getWikiLovesMonumentsYear(cal)))
    }
}